### DISCUSSION TITLE
using api to add existing pdf to database

### DISCUSSION THREAD

**Discussion by fab6** [papis/papis#837] [Q&A] [answered]
Hi,

similar as discussed here #836 I was thinking to add already downloaded pdfs via the api to the database.
For my understanding `papis.commands.add` could be used for this, which could then be combined with a separate python script to extract doi information from the pdf itself.
Do you have an example or can give me a hint how `papis.commands.add` can be used for this.
Thank you!
Fab

**Comment by jghauser** [comment, marked as answer]
Would e.g. `papis add FILE_PATH --from doi $(custom_script FILE_PATH)` do the trick? this could also be put in a script to avoid having to repeat `FILE_PATH`.

**Reply by fab6** [reply]
ok, yes, this should be possible
Thank you!
